spec/support/a_generic_event_endpoint.rb

```diff
@@ -13,17 +13,37 @@
       attributes: event_attributes,
       relationships: {
         eventable: { data: { type: eventable_type, id: eventable_id } },
-      },
+      }.merge(event_specific_relationships),
     }
   end
 
   let(:event_attributes) do
     attributes_for(factory).tap do |attributes|
       attributes.except!(:eventable)
       attributes[:event_type] = event_type
+      attributes[:details].keys.grep(/_id/).each do |relationship_key|
+        attributes[:details].delete(relationship_key)
+      end
     end
   end
 
+  let(:event_specific_relationships) do
+    attributes = attributes_for(factory)
+    relationship_attributes = attributes[:details].slice(*attributes[:details].keys.grep(/_id/))
+
+    relationship_attributes.each_with_object({}) do |(key, value), acc|
+      named_relationship_key = key.to_s.sub('_id', '')
+      acc[named_relationship_key] = {
+        'data' => {
+          'id' => value,
+          'type' => 'locations',
```

Comment:
is the type of those relationships always location?

Reply:
Currently, yeah. In future, if there are other types I can just override the invocation block for the shared example with a `let(:event_specific_relationships)` and they'll get merged in.